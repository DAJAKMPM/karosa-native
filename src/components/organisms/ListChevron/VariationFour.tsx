/**
 *
 * ListChevron
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import ListChevronStyles from "./styles";
import { View } from "react-native";

const VariantFour: FC<PropsType> = (props) => {
  const {
    hasBottomDivider,
    listColor,
    title,
    onPress,
    info,
    required,
    infoStyle,
  } = props;

  return (
    <ListItem bottomDivider={hasBottomDivider} onPress={onPress}>
      <ListItem.Content style={ListChevronStyles.container}>
        {listColor && (
          <MaterialCommunityIcons
            style={ListChevronStyles.listColor}
            name="checkbox-blank"
            color={listColor}
          />
        )}
        <Text
          text={title}
          textStyle={[
            listColor
              ? ListChevronStyles.txtListColor
              : ListChevronStyles.title,
          ]}
        />
        {required && (
          <Text text={"*"} textStyle={ListChevronStyles.txtRequired} />
        )}
      </ListItem.Content>
      {info && (
        <View style={ListChevronStyles.infoContainerWithoutChevron}>
          <Text
            text={info}
            textStyle={[ListChevronStyles.txtInfo, infoStyle]}
          />
        </View>
      )}
    </ListItem>
  );
};

export default VariantFour;
