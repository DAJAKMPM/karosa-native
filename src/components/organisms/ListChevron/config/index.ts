/**
 *
 * @format
 *
 */

export const FONT_SIZE = 28;
export const NUM_LINE = 1;
