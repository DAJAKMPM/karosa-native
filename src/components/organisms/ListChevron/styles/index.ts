/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme, DIMENS, SPACING } from "@app/styles";

const ListChevronStyles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
  },
  txtListColor: {
    ...theme.textRegular,
    position: "absolute",
    left: 35,
    fontWeight: "400",
  },
  listColor: {
    position: "absolute",
    left: 0,
    fontSize: 30,
  },
  icon: {
    position: "relative",
    left: 0,
  },
  title: {
    ...theme.textRegular,
    fontWeight: "400",
  },
  txtWithIcon: {
    ...theme.textListItem,
    position: "relative",
    left: 7,
    marginRight: 6,
    fontWeight: "400",
  },
  txtAvatarTitle: {
    ...theme.textLight,
    color: theme.colors.dark20,
    fontWeight: "400",
  },
  txtInfo: {
    ...theme.textLight,
    color: theme.colors.dark30,
    fontWeight: "300",
    textAlign: "right",
  },
  txtSubtitle: {
    ...theme.textLight,
    color: theme.colors.dark30,
    fontWeight: "300",
  },
  txtRequired: {
    ...theme.textRegular,
    marginLeft: SPACING.xs,
    color: theme.colors.red5,
    fontWeight: "400",
  },
  avatarImg: {
    width: DIMENS.screenWidth * 0.1,
    height: DIMENS.screenHeight * 0.056,
    borderRadius: 50,
  },
  infoContainer: {
    position: "relative",
    left: 12,
  },
  infoSubtitleContainer: {
    position: "relative",
    left: 12,
    alignSelf: "flex-start",
  },
  iconSubtitle: {
    alignSelf: "flex-start",
    top: -3,
  },
  txtSubtitleVariationOne: {
    ...theme.textRegular,
    color: theme.colors.dark30,
    position: "relative",
    left: 7,
    top: 4,
    fontWeight: "400",
  },
  chevronSubtitleContainer: {
    alignSelf: "flex-start",
    top: 2,
  },
  infoContainerWithoutChevron: {
    position: "relative",
    // left: 9,
  },
});

export default ListChevronStyles;
