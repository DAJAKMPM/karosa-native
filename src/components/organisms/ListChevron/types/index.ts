/**
 *
 * @format
 *
 */

 import { StyleProp, TextStyle } from "react-native";

export type PropsType = {
  title: string;
  variation?: number;
  listColor?: string;
  subtitle?: string;
  info?: string;
  chevronColor?: string;
  avatar?: string;
  hasBottomDivider?: boolean;
  required?: boolean;
  infoStyle?: StyleProp<TextStyle>;
  iconGroup?: string;
  iconName?: string;
  iconWidth?: number;
  iconHeight?: number;
  onPress: () => void;
  hasChevron?: boolean;
};
