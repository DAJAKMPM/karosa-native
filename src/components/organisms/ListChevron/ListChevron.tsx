/**
 *
 * ListChevron
 * @format
 *
 */

import React, { FC } from "react";

import type { PropsType } from "./types";
import VariationOne from "./VariationOne";
import VariationTwo from "./VariationTwo";
import VariationThree from "./VariationThree";
import VariationFour from "./VariationFour";

const ListChevron: FC<PropsType> = (props) => {
  const {
    hasBottomDivider,
    title,
    variation,
    onPress,
    subtitle,
    avatar,
    chevronColor,
    info,
    infoStyle,
    listColor,
    required,
    iconWidth,
    iconHeight,
    iconGroup,
    iconName,
    icon,
    containerStyle,
    titleStyle,
  } = props;

  switch (variation) {
    case 1:
      return (
        <VariationOne
          variation={variation}
          hasBottomDivider={hasBottomDivider}
          listColor={listColor}
          title={title}
          onPress={onPress}
          containerStyle={containerStyle}
          info={info}
          required={required}
          infoStyle={infoStyle}
          chevronColor={chevronColor}
          subtitle={subtitle}
          icon={icon}
          titleStyle={titleStyle}
        />
      );

    case 2:
      return (
        <VariationTwo
          variation={variation}
          hasBottomDivider={hasBottomDivider}
          title={title}
          onPress={onPress}
          subtitle={subtitle}
          avatar={avatar}
          chevronColor={chevronColor}
        />
      );
    case 3:
      return (
        <VariationThree
          variation={variation}
          hasBottomDivider={hasBottomDivider}
          title={title}
          onPress={onPress}
          subtitle={subtitle}
          avatar={avatar}
          chevronColor={chevronColor}
          iconName={iconName}
          iconGroup={iconGroup}
          iconHeight={iconHeight}
          iconWidth={iconWidth}
        />
      );
    case 4:
      return (
        <VariationFour
          variation={variation}
          hasBottomDivider={hasBottomDivider}
          listColor={listColor}
          title={title}
          onPress={onPress}
          info={info}
          required={required}
          infoStyle={infoStyle}
        />
      );
    default:
      return (
        <VariationOne
          variation={variation}
          hasBottomDivider={hasBottomDivider}
          listColor={listColor}
          title={title}
          onPress={onPress}
          info={info}
          required={required}
          infoStyle={infoStyle}
        />
      );
  }
};

export default ListChevron;
