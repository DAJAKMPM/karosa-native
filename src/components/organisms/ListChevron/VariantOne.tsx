/**
 *
 * ListChevron
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { theme } from "@app/styles";
import Text from "@app/atoms/Text";

import type { PropsType } from "./types";
import ListChevronStyles from "./styles";
import { View } from "react-native";

const VariantOne: FC<PropsType> = (props) => {
  const {
    hasBottomDivider,
    listColor,
    title,
    onPress,
    info,
    required,
    infoStyle,
    chevronColor = theme.colors.dark10,
  } = props;

  return (
    <ListItem bottomDivider={hasBottomDivider} onPress={onPress}>
      <ListItem.Content style={ListChevronStyles.container}>
        {listColor && (
          <MaterialCommunityIcons
            style={ListChevronStyles.listColor}
            name="checkbox-blank"
            color={listColor}
          />
        )}
        <Text
          text={title}
          textStyle={[
            listColor
              ? ListChevronStyles.txtListColor
              : ListChevronStyles.title,
          ]}
        />
        {required && (
          <Text text={"*"} textStyle={ListChevronStyles.txtRequired} />
        )}
      </ListItem.Content>
      {info && (
        <View style={ListChevronStyles.infoContainer}>
          <Text
            text={info}
            textStyle={[ListChevronStyles.txtInfo, infoStyle]}
          />
        </View>
      )}
      <ListItem.Chevron iconStyle={{ color: chevronColor }} />
    </ListItem>
  );
};

export default VariantOne;
