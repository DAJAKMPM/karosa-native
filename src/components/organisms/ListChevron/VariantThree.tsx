/**
 *
 * VariantTwo
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { theme } from "@app/styles";
import Icon from "@app/atoms/Icon";

import type { PropsType } from "./types";
import { FONT_SIZE } from "./config";
import ListChevronStyles from "./styles";

const VariantThree: FC<PropsType> = (props) => {
  const {
    hasBottomDivider,
    title,
    onPress,
    subtitle,
    iconGroup,
    iconName,
    iconWidth,
    iconHeight,
    chevronColor = theme.colors.dark10,
  } = props;

  const iconStyle = {
    color: chevronColor,
    fontSize: FONT_SIZE,
  };

  return (
    <ListItem bottomDivider={hasBottomDivider} onPress={onPress}>
      <Icon
        group={iconGroup || ""}
        name={iconName || ""}
        height={iconHeight || 50}
        width={iconWidth || 50}
      />
      <ListItem.Content>
        <ListItem.Title style={ListChevronStyles.txtSubtitle}>
          {title}
        </ListItem.Title>
        <ListItem.Subtitle style={ListChevronStyles.txtSubtitle}>
          {subtitle}
        </ListItem.Subtitle>
      </ListItem.Content>
      <ListItem.Chevron iconStyle={iconStyle} />
    </ListItem>
  );
};

export default VariantThree;
