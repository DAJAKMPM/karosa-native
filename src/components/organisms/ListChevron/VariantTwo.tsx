/**
 *
 * VariantTwo
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import { theme } from "@app/styles";
import Image from "@app/atoms/Image";

import { defaultFontSize } from "./config";
import type { PropsType } from "./types";
import ListChevronStyles from "./styles";

const VariantTwo: FC<PropsType> = (props) => {
  const {
    hasBottomDivider,
    title,
    onPress,
    subtitle,
    avatar,
    chevronColor = theme.colors.dark10,
  } = props;

  const iconStyle = {
    color: chevronColor,
    fontSize: defaultFontSize,
  };

  return (
    <ListItem bottomDivider={hasBottomDivider} onPress={onPress}>
      <Image
        source={{ uri: avatar }}
        customStyle={ListChevronStyles.avatarImg}
        resizeMode={"cover"}
      />
      <ListItem.Content>
        <ListItem.Title style={ListChevronStyles.txtAvatarTitle}>
          {title}
        </ListItem.Title>
        <ListItem.Subtitle style={ListChevronStyles.txtSubtitle}>
          {subtitle}
        </ListItem.Subtitle>
      </ListItem.Content>
      <ListItem.Chevron iconStyle={iconStyle} />
    </ListItem>
  );
};

export default VariantTwo;
