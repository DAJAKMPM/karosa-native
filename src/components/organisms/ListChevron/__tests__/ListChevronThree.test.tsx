import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import { COMMON } from "@app/constants";
import ListChevron from "../ListChevron";

describe("ListChevron variation three Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <ListChevron
        onPress={noop}
        title="test"
        iconGroup="products"
        iconName="chatNow"
        iconHeight={20}
        iconWidth={20}
        variation={COMMON.VARIATION.THREE}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
