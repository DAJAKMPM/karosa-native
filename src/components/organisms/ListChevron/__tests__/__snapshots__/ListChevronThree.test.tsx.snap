// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListChevron variation three Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View>
        <SvgMock
          height={20}
          width={20}
        />
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#4F4F4F",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 14.06,
            }
          }
          testID="listItemTitle"
        >
          test
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#4F4F4F",
              "fontSize": 12,
              "fontWeight": "300",
              "lineHeight": 14.06,
            }
          }
          testID="listItemTitle"
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
